libs/execution/execution.go

```diff
@@ -531,3 +533,15 @@ func cleanupTerraformPlan(nonEmptyPlan bool, planError error, stdout string, std
 func (d DiggerExecutor) projectId() string {
 	return d.ProjectNamespace + "#" + d.ProjectName
 }
+
+// this will log an exit code and error based on the executor of the executor drivers are by filename
+func logCommandFail(exitCode int, err error) {
+
+	_, filename, _, ok := runtime.Caller(1);
+	if ok {
+		executor := strings.TrimSuffix(path.Base(filename), path.Ext(filename))
+		log.Printf("Command failed in %v with exit code %v and error %v", executor, exitCode, err)
+	} else {
+		log.Printf("Command failed in unknown executor with exit code %v and error %v", exitCode, err)
+	}
+}
```

Comment:
thanks so much for this improvement, I remember terragrunt jobs always had obscure error messages especially when they failed